src/ord/map.rs

```diff
@@ -1906,6 +1908,46 @@ where
 
 impl<'a, K, V> ExactSizeIterator for Iter<'a, K, V> where (K, V): 'a + BTreeValue {}
 
+/// An iterator over a range of key/value pairs in a map.
+pub struct RangeIter<'a, K, V> {
+    // This uses a `RangedIter` internally, but we only construct it when the range is
+    // full, meaning that we can override `size_hint` and implement `ExactSizeIterator`.
```

Comment:
This is the same comment as for `Iter`, but there is no size hint override or `impl ExactSizeIterator` for this new type.